### ISSUE TITLE
Paginated search results only return 1 page

### ISSUE THREAD

**Issue by SimonBarrettACT** [shokme/laravel-scout-meilisearch#95]
Since updating to 1.2.3 (the issue is also present in 1.2.2. ), all searches using the following code only return 1 page.

Downgrading to 1.2.1 is necessary for the code to work correctly.

```
$courses = Course::_search_($this->searchString, function (Indexes $meilisearch, $query, $options) {  
        $options\['filters'\] = 'active=true';  
        return $meilisearch->search($query, $options);  
    })->paginate(12);
```

**Comment by curquiza** [comment]
Hello! That's my fault, it's because I removed the usage of `nbHits` in the code, which is not a trustable value since it's not an exhaustive value. `nbHits` is indeed not reliable for pagination because can be exhaustive or not, depending on the value of `exhaustiveNbHits` that MeiliSearch returns which is always `false` for the moment. So, it worked by chance before my PR.

However my PR makes things worst because not well implemented. The previous state was not perfect but better that the changes I introduced.
I don't have the time to remove correctly the nbHits usage it so I'm going to revert my PR, and I will fix it when I have time. Sorry for this.

We should implement the pagination as described here: https://github.com/meilisearch/documentation/issues/561

-----

FYI: The Meili team is aware of the confusion with `nbHis`. Here are the different issues and comments about it to explain why it's confusing, and why we should not use it:
- https://github.com/meilisearch/MeiliSearch/issues/1120
- https://github.com/meilisearch/documentation/issues/561
- https://github.com/meilisearch/meilisearch-php/issues/119#issuecomment-732908568

We are sorry for this. We all hope this confusion will be fixed asap in MeiliSearch.
